x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/form/deserializer_and_serializer.test.ts

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { setAutoFreeze } from 'immer';
+import { cloneDeep } from 'lodash';
+import { SerializedPolicy } from '../../../../../common/types';
+import { deserializer } from './deserializer';
+import { createSerializer } from './serializer';
+import { FormInternal } from '../types';
+
+const isObject = (v: unknown): v is { [key: string]: any } =>
+  Object.prototype.toString.call(v) === '[object Object]';
+
+const unknownValue = { some: 'value' };
+
+const populateWithUnknownEntries = (v: unknown) => {
+  if (isObject(v)) {
+    for (const key of Object.keys(v)) {
+      if (key === 'require' || key === 'include' || key === 'exclude') continue; // this will generate an invalid policy
```

Comment:
Thanks for adding this helpful comment! This might be a bit easier to read:

```js
if (['require', 'include', 'exclude'].includes(key)) continue;
```